util/opentelemetry-util-http/src/opentelemetry/util/http/httplib.py

```diff
@@ -104,8 +110,15 @@ def trysetip(conn: http.client.HTTPConnection, loglevel=logging.DEBUG) -> bool:
             stack_info=True,
         )
     else:
+        _OpenTelemetrySemanticConventionStability._initialize()
+        stability_mode = _OpenTelemetrySemanticConventionStability._get_opentelemetry_stability_opt_in_mode(
+            _OpenTelemetryStabilitySignalType.HTTP
+        )
         for span in spanlist:
-            span.set_attribute(SpanAttributes.NET_PEER_IP, ip)
+            if _report_old(stability_mode):
```

Comment:
You can use the helper function [here](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/opentelemetry-instrumentation/src/opentelemetry/instrumentation/_semconv.py#L313) instead.

Reply:
I have tried to use this one, but in `_set_http_peer_ip`, it use this function
```
def set_string_attribute(result, key, value):
    if value:
        result[key] = value
```

but span do not support set attributes like this, so I write it like this